Zend/zend_vm_def.h

```diff
@@ -4437,6 +4443,7 @@ ZEND_VM_INLINE_HANDLER(62, ZEND_RETURN, CONST|TMP|VAR|CV, ANY, SPEC(OBSERVER))
 	}
 	ZEND_OBSERVER_SAVE_OPLINE();
 	ZEND_OBSERVER_FCALL_END(execute_data, return_value);
+	ZEND_SIMPLE_INTERRUPT_CHECK(execute_data);
```

Comment:
No... Why did you introduce this check? Just for a yet another observer feature?
It's going to make slowdown. :(
JIT should consistently reflect all these VM changes as well. :(

Instead you might check ``EG(vm_interrupt)`` directly in ``zend_observer_fcall_end()``, but it's inlined anyway.

Reply:
It seemed better to have consistent rules on when interrupts are triggered, and also made the code simpler. I can pursue other options, but know that certain code will definitely get uglier.

Reply:
My last comment wasn't precise enough. I meant if we keep interrupts in roughly all the same places as on master (except for the change of making the internal one happen while the internal frame is still on the stack), then it will be uglier. If we're fine dropping some cases, then it's not necessarily worse. For instance, on master we can call the interrupt handler if `zend_execute_ex` has been set.

Reply:
Interrupts are introduced to perform a "serialized" break of interpretation of a long running script with minimal number of checks. We check interrupts only in 3 places - loops, recursive calls, returns from internal functions.  This handles possible "infinity" loops and long running internal functions.

There is no reason to check interrupts in RETURN, because long-running user functions should be already caught by loop/call checks.